GitUI/CommandsDialogs/BrowseDialog/FormRecentReposSettings.cs

```diff
@@ -18,14 +18,10 @@ public FormRecentReposSettings()
             InitializeComponent();
             InitializeComplete();
 
-            ThreadHelper.JoinableTaskFactory.Run(async () =>
-            {
-                _repositoryHistory = await RepositoryHistoryManager.Locals.LoadRecentHistoryAsync();
-
-                await this.SwitchToMainThreadAsync();
-                LoadSettings();
-                RefreshRepos();
-            });
+            _repositoryHistory = ThreadHelper.JoinableTaskFactory.Run(RepositoryHistoryManager.Locals.LoadRecentHistoryAsync);
+            ThreadHelper.AssertOnUIThread();
```

Comment:
The assert should be in `LoadRecentHistoryAsync`, not here (the called methods could be refactored).
(Just omitting here (possibly with a comment) is OK).
Not commenting on all occurrences below.

Reply:
> The assert should be in `LoadRecentHistoryAsync`, not here (the called methods could be refactored).

`LoadRecentHistoryAsync` itself does *not* need to be run on the UI thread. It switches to a background thread unconditionally. (Since it is never called asynchronously, async is an unnecessary overhead.)
In a follow-up PR, `JTF.Run()` and initial `await TaskScheduler.Default;` will only be done by `ThreadHelper/TaskManager.FileAndForget()`.

Reply:
Then remove the assertion?

Reply:
> The assert should be in `LoadRecentHistoryAsync`

Thread affinity assertions should not be placed inside asynchronous methods, since they can (and are expected to) simply switch to the necessary context. See https://github.com/gitextensions/gitextensions/pull/11137#discussion_r1291661950.